explainer.md

```diff
@@ -215,6 +243,162 @@ third-party then signs either the exchanges in a package or the package as a
+inside it, and the client follows the remaining steps only if the Service Worker
+responds with an encoded `application/signed-exchange` resource. The
+`application/signed-exchange` also participates in caches the same way as any
+other resource.
+
+### Fetch the certificate chain
+
+The client parses the beginning of the `application/signed-exchange` resource to
+extract the [`Signature`
+header](https://wicg.github.io/webpackage/draft-yasskin-http-origin-signed-responses.html#signature-header).
+Each signature in the `Signature` header has a `certUrl` field that identifies a
+certificate chain to use to validate the signature, and each certificate chain
+is fetched and validated.
+
+The request for this certificate chain is made without credentials and skips Service
+Workers, for compatibility with browser network stack architectures. (This may
+change if network stacks prove to be less of a problem than expected.) The
+certificate is cached as a normal HTTP request.
+
+The client checks the leaf certificate's hash against the `certSha256` field of
+the `Signature` header and then validates the certificate chain. Certificate
+validation is a call to an OS-controlled library and may make further URL
+requests that aren't under the browser's control and therefore use a separate,
+probably ephemeral, cookie jar, cache, and set of service workers.
+
+Each signature with a valid certificate chain is passed on to the next step.
+
+### Signature verification
+
+Once the certificates are validated and enough of the
+`application/signed-exchange` resource is received to parse the claimed
+signed-exchange headers, the client extracts the logical URL from those headers
+and then tries to find a valid signature over the headers that is trusted for
+the claimed origin. If it can't find any, it either
+
+1. redirects to the logical URL as if the whole signed exchange were a 302
+   response, or
+2. fails with a network error.
+
+We're not yet certain which behavior is best. The first is slightly more
+resilient to clock skew, while the second encourages intermediates to get their
+implementations right.
+
+If the client does find a valid signature, it "releases" the headers for further
+processing and starts validating
+[mi-sha256](https://tools.ietf.org/html/draft-thomson-http-mice-02#section-2)
+records into a response stream as they arrive.
+
+### Prefetching stops here
+
+At this point, the client's behavior depends on whether the signed exchange was
+requested as a [prefetch](https://w3c.github.io/resource-hints/#prefetch). To
+satisfy the [privacy-preserving prefetch](#privacy-preserving-prefetch) use
+case, prefetches can't fully load the logical URL, which would create an HTTP
+cache entry that would be visible to the logical URL's server.
+
+Prefetches can and should process any `Link: <>; rel=preload` headers they find,
+as prefetches. If those point at signed exchanges, this process repeats.
+
+### Caching the signed response
```

Comment:
Let me clarify... at this point we start talking about caching the response(s) in the signed exchange, is that right?

Reply:
Heh, yes, that wasn't clear at all. Is this new wording better?

Reply:
Loos a lot clearer, thanks!